stdpopsim/slim_engine.py

```diff
@@ -1176,47 +1210,64 @@ def matrix2str(
         return "".join(s)
 
     # Genomic element types.
+    mutation_callbacks = {}
     dfe_mtypes = _dfe_to_mtypes(contig)
     for j, d, ints in _enum_dfe_and_intervals(contig):
-        # Mutation types.
+        # Mutation types and proportions.
         mut_type_list = []
+        mut_props_list = []
         assert len(dfe_mtypes[j]) == len(d.mutation_types)
-        for mid, m in dfe_mtypes[j]:
-            if len(m.Q_scaled_index) >= 1:
-                distrib_args = [str(arg) for arg in m.distribution_args]
-                for k in m.Q_scaled_index:
-                    distrib_args[m.Q_scaled_index[k]] = (
-                        "Q * " + distrib_args[m.Q_scaled_index[k]]
+        for mt_index, (mid_list, mt) in enumerate(dfe_mtypes[j]):
+            if len(mt.Q_scaled_index) >= 1:
+                distrib_args = [str(arg) for arg in mt.distribution_args]
+                for k in mt.Q_scaled_index:
+                    distrib_args[mt.Q_scaled_index[k]] = (
+                        "Q * " + distrib_args[mt.Q_scaled_index[k]]
                     )
                 distrib_args = ", ".join(distrib_args)
             # dealing with distributions given by "s" Eidos script,
             else:
-                distrib_args = "; ".join([f'"{a}"' for a in m.distribution_args])
-            mut_type_list.append(mid)
-            printsc(
-                f"    initializeMutationType({mid}, {m.dominance_coeff}, "
-                f'"{m.distribution_type}", {distrib_args});'
-            )
-            if not m.convert_to_substitution:
-                # T is the default for WF simulations.
-                printsc(f"    m{mid}.convertToSubstitution = F;")
-            # TODO: when msprime.SLiMMutationModel supports stacking policy,
-            # set policy such that there's at most a single mutation per-site
-            # and individual
-            # printsc(f"    m{mid}.mutationStackGroup = 0;")
-            # printsc(f"    m{mid}.mutationStackPolicy = 'l';")
-        mut_types = ", ".join([str(mt) for mt in mut_type_list])
+                distrib_args = "; ".join([f'"{a}"' for a in mt.distribution_args])
+            if mt.dominance_coeff_list is None:
+                h_list = [mt.dominance_coeff]
+            else:
+                h_list = [0.5]  # this value will not matter
```

Comment:
is None or -1 or something better?

Reply:
None will cause an error. And, if it's like `-1` or something weird then I can imagine people looking at that value and thinking "whoa something is wrong" even though it's not?  (Since the value *will* appear in the SLiM script, just for mutations that are never kept around.)

Reply:
I'll add a more clarifying comment.